node/tests/utils.rs

```diff
@@ -478,45 +478,43 @@ impl MASQNode {
         args
     }
 
-    fn virtual_arg_pairs(args: Vec<String>) -> Vec<(String, Option<String>)> {
-        fn look_at_val_distanced_by_two<I>(
-            peekable: &mut Peekable<I>,
-            current_string: String,
-            mut acc: Vec<(String, Option<String>)>,
-        ) -> Vec<(String, Option<String>)>
-        where
-            I: Iterator<Item = String>,
-        {
-            match peekable.peek() {
-                //possibly value for that arg
-                Some(next_string) => {
-                    if next_string.starts_with("--") {
-                        acc.push((current_string, None));
-                        acc
-                    } else {
-                        let value = peekable.next().unwrap();
-                        acc.push((current_string, Some(value)));
-                        acc
+    fn virtual_arg_pairs_with_remembered_position(
+        args: Vec<String>,
+    ) -> Vec<(String, PositionedArg)> {
+        let shifted_by_one = args
+            .clone()
+            .into_iter()
+            .map(|val| Some(val))
+            .skip(1)
+            .chain(once(None));
+        let params_and_yet_suspected_arg = args
+            .into_iter()
+            .zip(shifted_by_one.enumerate())
+            .filter(|(param, _)| param.starts_with("--"));
+        params_and_yet_suspected_arg
+            .map(|(param_name, (original_position, suspected_arg_opt))| {
+                let construct_positioned_arg = |val_opt| {
+                    (
+                        param_name,
+                        PositionedArg {
+                            arg_opt: val_opt,
+                            remembered_position: original_position,
+                        },
+                    )
+                };
+
+                match suspected_arg_opt {
+                    Some(suspected_arg) => {
+                        if suspected_arg.starts_with("--") {
+                            construct_positioned_arg(None)
+                        } else {
+                            construct_positioned_arg(Some(suspected_arg))
+                        }
                     }
+                    None => construct_positioned_arg(None),
```

Comment:
Comment about how this handles a parameter without a value specified at the end of the list.